### PR TITLE
python3Packages.google_auth: 1.20.1 -> 1.22.1

### PR DESCRIPTION
**Pull request by austinbutler**
###### Motivation for this change

Upgrades `google_auth`, fixes many Google Cloud packages. Will backport to 20.09 for ZHF.

https://hydra.nixos.org/build/127647323

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by austinbutler** [comment]
As I figured it might, this package requiring an updated `google_auth` opens a whole can of worms. That said, it's almost entirely various `google-cloud` Python packages so doesn't seem insurmountable. I'll see if I can tackle it, but if there's a particular way such a process needs to go please let me know.

**Comment by austinbutler** [comment]
All right, worked through all the issues except a Tensorflow package that doesn't build on master anyways.

**Comment by austinbutler** [comment]
@jonringer @costrouc mind taking a look? Also now that 20.0 is released would I still backport this?

**Comment by jonringer** [comment]
> @jonringer @costrouc mind taking a look? Also now that 20.09 is released would I still backport this?

you may have to remove some "broken = true;", but yea. I'm fine with backporting non-major bumps. And if it doesn't cause regressions.

**Comment by jonringer** [comment]
only one I'm concerned about is:
```
python3Packages.google_cloud_automl: 1.0.1 -> 2.0.0
```

File: pkgs/development/python-modules/google_cloud_bigquery/default.nix [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
+    google_cloud_core
+    pandas
+    proto-plus
+    pyarrow
+  ];
+
+  # prevent google directory from shadowing google imports
   # call_api_applying_custom_retry_on_timeout requires credentials
   # test_magics requires modifying sys.path
   checkPhase = ''
```
Comment (jonringer) [inline pkgs/development/python-modules/google_cloud_bigquery/default.nix:29]: maybe disable this test?

```
builder for '/nix/store/2nm6nj24sswksj2p5gw7myyddpbywg4l-python3.8-google-cloud-bigquery-2.1.0.drv' failed with exit code 1; last 10 log lines:
  E       datetime.datetime(4567, 1, 1, 0, 0)
  E
  E       + [datetime.datetime(4567, 1, 1, 0, 0), datetime.datetime(9999, 12, 31, 0, 0)]
  E       - [datetime.datetime(4567, 1, 1, 0, 0, tzinfo=<UTC>),
  E       -  datetime.datetime(9999, 12, 31, 0, 0, tzinfo=<UTC>)]

  tests/unit/test_table.py:2294: AssertionError
  =========================== short test summary info ============================
  FAILED tests/unit/test_table.py::TestRowIterator::test_to_dataframe_timestamp_out_of_pyarrow_bounds
  ========== 1 failed, 1372 passed, 54 skipped, 1 deselected in 14.60s ===========
```

Comment (jonringer) [reply pkgs/development/python-modules/google_cloud_bigquery/default.nix:29]: `pytestCheckHook` would make this easy

Comment (austinbutler) [reply pkgs/development/python-modules/google_cloud_bigquery/default.nix:29]: Strange, all tests pass on my machine. Just checked again with `nix-build -A python38Packages.google_cloud_bigquery`. I can disable the test, though.

**Comment by austinbutler** [comment]
> > @jonringer @costrouc mind taking a look? Also now that 20.09 is released would I still backport this?
> 
> you may have to remove some "broken = true;", but yea. I'm fine with backporting non-major bumps. And if it doesn't cause regressions.

Not sure what you mean. I didn't explicitly add any `broken = true` at least. This is probably my own lack of knowledge.

> only one I'm concerned about backporting:
> 
> ```
> python3Packages.google_cloud_automl: 1.0.1 -> 2.0.0
> ```

Are you still evaluating or do you want me to look into if it's possible to keep this on 1.x?

**Comment by jonringer** [comment]
> Not sure what you mean. I didn't explicitly add any `broken = true` at least. This is probably my own lack of knowledge.

I did on release-20.09, as part of the release https://github.com/NixOS/nixpkgs/pull/101736

**Comment by jonringer** [comment]
> Are you still evaluating or do you want me to look into if it's possible to keep this on 1.x?

For backporting, I would prefer not to backport this, unless it's already broken

File: pkgs/development/python-modules/google_cloud_bigquery/default.nix (hunk 1 above)
Comment (austinbutler) [reply pkgs/development/python-modules/google_cloud_bigquery/default.nix]: Disabled.

**Comment by austinbutler** [comment]
> > Are you still evaluating or do you want me to look into if it's possible to keep this on 1.x?
> 
> For backporting, I would prefer not to backport this, unless it's already broken

Just checked and this one was not broken on 20.09. So I will keep that in mind when backporting.

**Comment by austinbutler** [comment]
@jonringer would you mind taking another look? I think this is good to go.

**Comment by austinbutler** [comment]
@jonringer pretty please? 🙇

**Comment by jonringer** [comment]
oh yea, sorry, last time it was just taking longer than expected to review these packages

**Review by jonringer — APPROVED** [review]
LGTM

failures are broken on target branch
```
https://github.com/NixOS/nixpkgs/pull/101029
2 packages marked as broken and skipped:
kmymoney python37Packages.rl-coach

1 package failed to build:
python37Packages.tensorflow-bin_2

118 packages built:
beancount deja-dup duplicity duply dvc-with-remotes fava gcalcli ledger2beancount lieer patroni python37Packages.beancount python37Packages.beancount_docverif python37Packages.cirq python37Packages.goobook python37Packages.google-auth-httplib2 python37Packages.google-auth-oauthlib python37Packages.google-cloud-access-context-manager python37Packages.google-cloud-iam python37Packages.google-cloud-org-policy python37Packages.google_api_core python37Packages.google_api_python_client python37Packages.google_auth python37Packages.google_cloud_asset python37Packages.google_cloud_automl python37Packages.google_cloud_bigquery python37Packages.google_cloud_bigquery_datatransfer python37Packages.google_cloud_bigtable python37Packages.google_cloud_container python37Packages.google_cloud_core python37Packages.google_cloud_dataproc python37Packages.google_cloud_datastore python37Packages.google_cloud_dlp python37Packages.google_cloud_dns python37Packages.google_cloud_error_reporting python37Packages.google_cloud_firestore python37Packages.google_cloud_iot python37Packages.google_cloud_kms python37Packages.google_cloud_language python37Packages.google_cloud_logging python37Packages.google_cloud_monitoring python37Packages.google_cloud_pubsub python37Packages.google_cloud_redis python37Packages.google_cloud_resource_manager python37Packages.google_cloud_runtimeconfig python37Packages.google_cloud_secret_manager python37Packages.google_cloud_securitycenter python37Packages.google_cloud_spanner python37Packages.google_cloud_speech python37Packages.google_cloud_storage python37Packages.google_cloud_tasks python37Packages.google_cloud_testutils python37Packages.google_cloud_texttospeech python37Packages.google_cloud_trace python37Packages.google_cloud_translate python37Packages.google_cloud_videointelligence python37Packages.google_cloud_vision python37Packages.google_cloud_websecurityscanner python37Packages.gspread python37Packages.kubernetes python37Packages.proto-plus python37Packages.pydrive python37Packages.tensorflow-tensorboard_2 python37Packages.weboob python38Packages.beancount_docverif python38Packages.cirq python38Packages.goobook python38Packages.google-auth-httplib2 python38Packages.google-auth-oauthlib python38Packages.google-cloud-access-context-manager python38Packages.google-cloud-iam python38Packages.google-cloud-org-policy python38Packages.google_api_core python38Packages.google_api_python_client python38Packages.google_auth python38Packages.google_cloud_asset python38Packages.google_cloud_automl python38Packages.google_cloud_bigquery python38Packages.google_cloud_bigquery_datatransfer python38Packages.google_cloud_bigtable python38Packages.google_cloud_container python38Packages.google_cloud_core python38Packages.google_cloud_dataproc python38Packages.google_cloud_datastore python38Packages.google_cloud_dlp python38Packages.google_cloud_dns python38Packages.google_cloud_error_reporting python38Packages.google_cloud_firestore python38Packages.google_cloud_iot python38Packages.google_cloud_kms python38Packages.google_cloud_language python38Packages.google_cloud_logging python38Packages.google_cloud_monitoring python38Packages.google_cloud_pubsub python38Packages.google_cloud_redis python38Packages.google_cloud_resource_manager python38Packages.google_cloud_runtimeconfig python38Packages.google_cloud_secret_manager python38Packages.google_cloud_securitycenter python38Packages.google_cloud_spanner python38Packages.google_cloud_speech python38Packages.google_cloud_storage python38Packages.google_cloud_tasks python38Packages.google_cloud_testutils python38Packages.google_cloud_texttospeech python38Packages.google_cloud_trace python38Packages.google_cloud_translate python38Packages.google_cloud_videointelligence python38Packages.google_cloud_vision python38Packages.google_cloud_websecurityscanner python38Packages.gspread python38Packages.kubernetes python38Packages.proto-plus python38Packages.pydrive python38Packages.tensorflow-tensorboard_2 python38Packages.weboob s3ql tuijam wal_e
```

**Comment by austinbutler** [comment]
> oh yea, sorry, last time it was just taking longer than expected to review these packages

All good, was just making me anxious since it was a large amount of packages and already had to merge in new changes once or twice. Thank you!

**Pull request merged**